src/App.tsx

```diff
@@ -1,41 +1,50 @@
-import React from 'react';
 import './App.scss';
+import { Component } from 'react';
+import { Clock } from './components/Clock';
 
-function getRandomName(): string {
-  const value = Math.random().toString().slice(2, 6);
+type State = {
+  clockName: string,
+  hasClock: boolean,
+};
 
-  return `Clock-${value}`;
-}
+class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    hasClock: true,
+    clockName: 'Clock 1',
+  };
+
+  componentDidMount() {
+    document.addEventListener('click', this.visibleClock);
+    document.addEventListener('contextmenu', this.hiddenClock);
+  }
 
-const App: React.FC = () => {
-  const date = new Date();
-  const clockName = getRandomName();
+  setRandomName = setInterval(() => {
```

Comment:
In case you stored Id in variable, you should call clearInterval function in `ComponentWillUnmount` to avoid memory leak
```suggestion
  timerId = setInterval(() => {
```